Test/FMI3/fmu_dummy/fmu3_model.c

```diff
@@ -110,35 +110,44 @@ fmi3Status fmi_set_debug_logging(fmi3Instance instance, fmi3Boolean loggingOn)
     } else {
         inst->loggingOn = loggingOn;
         return fmi3OK;
-    }
-}
+    }}
 
-fmi3Status fmi_get_float64(fmi3Instance instance, const fmi3ValueReference vr[], size_t nvr, fmi3Float64 value[], size_t nValues)
+fmi3Status fmi_get_float64(fmi3Instance instance, const fmi3ValueReference valueReferences[],
+    size_t nValueReferences, fmi3Float64 values[], size_t nValues)
 {
     instance_ptr_t inst = instance;
     if (inst == NULL) {
         return fmi3Fatal;
     } else {
-        size_t k;
-        size_t m = 0; /* index in value, considering that arrays will require many indices */
-        for (k = 0; k < nvr; k++) {
-            fmi3ValueReference cvr = vr[k];
-            if (cvr < N_STATES) {
-                value[m] = inst->states[cvr];
+        int k;
+        int m = 0; /* index in values, considering that arrays will require many indices */
+        for (k = 0; k < nValueReferences; k++) {
+            fmi3ValueReference currentValueReference = valueReferences[k];
+            if (currentValueReference < N_STATES) {
+                values[m] = inst->states[currentValueReference];
             }
-            else if(cvr == 4) {
+            else if(currentValueReference == 4) {
                 calc_get_derivatives(inst);
-                value[m] = inst->states_der[1];
+                values[m] = inst->states_der[1];
             }
-            else if (cvr == 12) { /* special case: array */
+            else if (currentValueReference == 12) { /* special case: array */
                 calc_get_derivatives(inst);
-                value[m++] = inst->states[0];
-                value[m++] = inst->states_der[0];
-                value[m++] = inst->states[1];
-                value[m++] = inst->states_der[1];
+                values[m++] = inst->states[0];
+                values[m++] = inst->states_der[0];
+                values[m++] = inst->states[1];
+                values[m++] = inst->states_der[1];
+            }
+            else if (currentValueReference == 1234567) {
+                /* Obviously values reference -1 does not exist but this enables
+                    easy testing of get/set.
+                 */
+                for(int i = 0; i < 4; i++) {
+                    values[i] = inst->dummy_array[i];
+                    m++;
```

Comment:
This shouldn't be updated, right?

Reply:
If it doesn't it will return fmi3Error because of the check in the end of the function. I am not sure why the check is done in the first place. Can I remove it?

Reply:
I removed it because I have removed the float 64 test. See commit d186a78.